src/node/net.js

```diff
@@ -0,0 +1,1109 @@
+  },
+  bufferSize: {
+    __proto__: null,
+    get() {
+      if (this._handle) {
+        return this.writableLength;
+      }
+    },
+  },
+  [kUpdateTimer]: {
+    __proto__: null,
+    get() {
+      return this._unrefTimer;
+    },
+  },
+  bytesRead: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() {
+      return this._handle ? this._handle.bytesRead : this[kBytesRead];
+    }
+  },
+  bytesWritten: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() {
+      const flushed = this._handle != null ? this._handle.bytesWritten : this[kBytesWritten];
+      const pending = this._writableState != null ? this._writableState.length : 0;
+      return flushed + pending;
+    }
+  },
+  remoteAddress: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() {
+      return this._getpeername().address;
+    }
+  },
+  remoteFamily: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() {
+      return this._getpeername().family;
+    }
+  },
+  remotePort: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() {
+      return this._getpeername().port;
+    }
+  },
+  localAddress: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() { return '0.0.0.0'; }
+  },
+  localPort: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() { return 0; }
+  },
+  localFamily: {
+    __proto__: null,
+    configurable: false,
+    enumerable: true,
+    get() { return 'IPv4'; }
+  },
+});
+
+// ======================================================================================
+// Helper/utility methods
+
+function cleanupAfterDestroy(socket, cb, error) {
+  if (socket._handle != null) {
+    socket[kBytesRead] = socket.bytesRead;
+    socket[kBytesWritten] = socket.bytesWritten;
+  }
+  socket._handle = null;
+  socket[kLastWriteQueueSize] = 0;
+  socket[kBuffer] = null;
+  socket[kBufferCb] = null;
+  socket[kBufferGen] = null;
+  socket[kSocketInfo] = null;
+  cb(error);
+  queueMicrotask(() => socket.emit('close', error != null));
+}
+
+function initializeConnection(socket, options) {
+  // options.localAddress, options.localPort, and options.family are ignored.
+  let {
+    host = 'localhost',
+    port = 0,
+    family,
+    hints,
+    autoSelectFamily,
+    lookup,
+  } = options;
+
+  if (!!autoSelectFamily) {
+    // We don't support this option. If the value is falsy, we can safely ignore it.
+    // If the value is truthy, we'll throw an error.
+    throw new ERR_INVALID_ARG_VALUE('options.autoSelectFamily',
+                                    autoSelectFamily,
+                                    'is not supported');
+  }
+
+  if (typeof port !== 'number' && typeof port !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('options.port', ['number', 'string'], port);
+  }
+  port = validatePort(+port);
```

Comment:
The `+` casting and type checks are performed in `validatePort`, so doing them here seems redundant. Though I guess it's fine for better error messages.